### PR TITLE
Closes AgileVentures/MetPlus_tracker#235

### PR DESCRIPTION
**Pull request by patmbolger**
All features for managing 'Skills' in the agency admin page: add, edit, update, delete.

### CODE REVIEW

File: app/controllers/agency_admin_controller.rb [hunk 1]
```diff
@@ -25,13 +25,34 @@ def home
...
+      when 'skills'
+        @skills = Skill.order(:name).
+                    page(params[:skills_page]).per_page(10)
+
+        render partial: 'job_properties', object: @skills,
+                locals: {data_type:  'skills',
+                         partial_id: 'skills_table',
+                         edit_property_path:   :edit_skill_path,
+                         delete_property_path: :skill_path}
+
```
Comment (joaopapereira) [inline app/controllers/agency_admin_controller.rb]: Shouldn't we render error when we do not know the data_type?

File: app/controllers/skills_controller.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+class SkillsController < ApplicationController
+  def create
+    skill = Skill.new(skill_params)
```
Comment (joaopapereira) [inline app/controllers/skills_controller.rb]: Is this a Ajax request?

**Comment by joaopapereira** [comment]
Once again this is a Huge change, and looks like it as things that are not related with this Story. Due to the size of the story I cannot judge it all by myself so I need someone else to review the code.

**Comment by joaopapereira** [comment]
Another thing that I just saw, this story isn't even in the scope for the current sprint........

**Comment by patmbolger** [comment]
Actually, everything in this PR is related to the story.

I used my discretion to move it into this sprint.  I can discuss at today's scrum.

File: app/controllers/skills_controller.rb (hunk 2 above)
Comment (patmbolger) [inline app/controllers/skills_controller.rb]: Yes.  I added a filter to confirm that only XHR requests are sent to the controller.

File: app/controllers/agency_admin_controller.rb (hunk 1 above)
Comment (patmbolger) [inline app/controllers/agency_admin_controller.rb]: Added an exception if data_type is not recognized.

**Comment by joaopapereira** [comment]
I assume this is correct and if no one add any comment till tomorrow afternoon I will merge this PR

**Pull request merged**